### ISSUE TITLE
Add Additional Common Resolutions to Aspect Ratio Bucketing

### ISSUE THREAD

**Issue by Eugeoter** [d8ahazard/sd_dreambooth_extension#1089]
**WHAT**
Aspect Ratio Bucketing. The aspect ratio bucketing feature does offer a bucket with max_resolution as the maximum of the width or height. However, if my training set contains both common resolutions such as 512x512, 768x768, and even 640x640, setting max_resolution to 768 will result in lower-resolution images with the same aspect ratio (1:1) being cropped and scaled into rectangles rather than squares, leading to loss of image information. For example, 512x512->384x768. However, 512x512 and 640x640 are two common resolutions used for training. Therefore, it is desirable to add these resolutions in a special way.

**HOW**
I simply add some lines into make_bucket_resolutions function in image_utils.py. It works.

```python
if max_resolution > 512 and (512, 512) not in resos:
        resos.add((512, 512))
if max_resolution > 640 and (640, 640) not in resos:
        resos.add((640, 640))
if max_resolution > 768 and (768, 768) not in resos:
        resos.add((640, 640))
```

**Comment by ArrowM** [comment]
I think `closest_resolution()` was mishandling images that require upscaling. Please checkout the dev branch and see if that fixes your issue.

**Comment by ArrowM** [comment]
With the correction made to dev and this one to optimizers 9b5447e7cc06c4c7a3d4481c1dcc109ba61ce9bc (plus the lack of response), I am marking this as closed.
